cypress/integration/control_attach.js

```diff
@@ -0,0 +1,89 @@
+context('Date Control', () => {
+	before(() => {
+		cy.login();
+		cy.visit('/app/doctype');
+		return cy.window().its('frappe').then(frappe => {
+			return frappe.xcall('frappe.tests.ui_test_helpers.create_doctype', {
+				name: 'Test Attach Control',
+				fields: [
+					{
+						"label": "Attach File or Image",
+						"fieldname": "attach",
+						"fieldtype": "Attach",
+						"in_list_view": 1,
+					},
+				]
+			});
+		});
+	});
+	it('Checking functionality for "Link" button in the "Attach" fieldtype', () => {
+		//Navigating to the new form for the newly created doctype
+		cy.new_form('Test Attach Control');	
+
+		//Clicking on the attach button which is displayed as part of creating a doctype with "Attach" fieldtype
+		cy.get('.control-input > .btn-sm').contains('Attach').click();
+
+		//Clicking on "Link" button to attach a file using the "Link" button
+		cy.get('.mt-2').findByRole('button', {name: 'Link'}).click();
+		cy.get('.input-group > .form-control').type('https://wallpaperplay.com/walls/full/8/2/b/72402.jpg');
+
+		//Clicking on the Upload button to upload the file
+		cy.get('.modal-footer > .standard-actions > .btn-primary').contains('Upload').click({delay: 500});
+		cy.wait(800);
```

Comment:
why?

wait for a specific request (like upload) instead.

Reply:
its because after saving the data, its takes time for the UI to update the change and the next line on the script fails if the changes are not getting reflected on the UI.

Reply:
> its because after saving the data, its takes time for the UI to update the change and the next line on the script fails if the changes are not getting reflected on the UI.

understood... 

All am suggesting is you can wait for the request (that upload button triggers) to complete instead of waiting for an arbitrary duration.

check [this code for "wait"](https://github.com/frappe/frappe/blob/ebd756c1e54ee2d988d995aa4c7e80373481259b/cypress/integration/control_link.js#L158) for reference.

Reply:
Corrected the same